src/python/pants/core/goals/run.py

```diff
@@ -132,8 +132,8 @@ def __init__(
         object.__setattr__(self, "digest", digest)
         object.__setattr__(self, "args", tuple(args))
         object.__setattr__(self, "extra_env", FrozenDict(extra_env or {}))
-        object.__setattr__(self, "immutable_input_digests", immutable_input_digests)
-        object.__setattr__(self, "append_only_caches", append_only_caches)
+        object.__setattr__(self, "immutable_input_digests", FrozenDict(immutable_input_digests))
+        object.__setattr__(self, "append_only_caches", FrozenDict(append_only_caches))
```

Comment:
This was likely cruft from removing `frozen_after_init`

Reply:
to avoid superflous object creation in case the inputs already are frozen dicts could do
```suggestion
        object.__setattr__(self, "immutable_input_digests", FrozenDict.freeze(immutable_input_digests))
        object.__setattr__(self, "append_only_caches", FrozenDict.freeze(append_only_caches))
```

Reply:
This wouldn't mirror the `FrozenDict` call above. I'd rather be consistent locally and we address `.freeze` across the board.